x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/searchablesnapshots/SearchableSnapshotShardStats.java

```diff
@@ -395,4 +396,55 @@ public int hashCode() {
             return Objects.hash(count, total, min, max);
         }
     }
+
+    public static class TimedCounter extends Counter {
+
+        private final long totalNanoseconds;
+
+        public TimedCounter(long count, long total, long min, long max, long totalNanoseconds) {
+            super(count, total, min, max);
+            this.totalNanoseconds = totalNanoseconds;
+        }
+
+        TimedCounter(StreamInput in) throws IOException {
+            super(in);
+            totalNanoseconds = in.readZLong();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeZLong(totalNanoseconds);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            {
+                builder.field("count", getCount());
+                builder.field("sum", getTotal());
+                builder.field("min", getMin());
+                builder.field("max", getMax());
+                builder.field("time", TimeValue.timeValueNanos(totalNanoseconds));
```

Comment:
`builder.timeField(String, String, long)` does not produce what we want here?

Reply:
No, I don't think so, this formats an absolute time in milliseconds-since-the-unix-epoch as a date, whereas here we have a relative time in nanoseconds.

It did, however, highlight that I should respect the `?human` parameter -- see 5b26f66ad3b.

Reply:
Right, I just checked the code. Thanks for supporting the human readable parameter.